src/reanimated2/UpdateProps.js

```diff
@@ -1,7 +1,8 @@
-/* global _updateProps _updatePropsJS */
+/* global _updateProps */
 import { processColor } from './Colors';
 import { makeShareable } from './core';
 import { Platform } from 'react-native';
+import { _updatePropsJS } from './js-reanimated/index';
```

Comment:
This will load the JS version regardless if it's needed or not. Maybe we should change [those](https://github.com/software-mansion/react-native-reanimated/pull/1482/files#diff-109d22d762ccae27889a4992c5797583a225c263244aa7dcc4d2df50415ca608R40-R41) to something along with this:

```
const updatePropsInternal = _updateProps ?? require('./js-reanimated/index'). _updatePropsJS
```
to load it on demand?

(`._updatePropsJS` instead of `?._updatePropsJS` because `_updatePropsJS` should be available at all times, so safe navigation there is not needed.)

Reply:
you can give it a shot but I've tried something similar some time ago and it didn't work. I think the proper solution for conditional imports is to put them in the files with certain extensions(like `.web.js`, `ios.js`, `android.js`)

reference: https://reactnative.dev/docs/platform-specific-code#platform-specific-extensions
example in our repo: https://github.com/software-mansion/react-native-reanimated/tree/master/src/reanimated2/platform-specific

I agree it should be imported only if needed 👍

Reply:
This generates a lot of files if want to keep it modularized.

If it doesn't work then we can stick to the older version, this change is something we should in mind though.